clients/ts/signalr/tests/HubConnection.test.ts

```diff
@@ -273,6 +273,27 @@ describe("HubConnection", () => {
             });
         });
 
+        it.only("start completes if connection closes and handshake not received yet", async () => {
+            await VerifyLogger.run(async (logger) => {
+                const mockProtocol = new TestProtocol(TransferFormat.Text);
+
+                const connection = new TestConnection(false);
+                const hubConnection = createHubConnection(connection, logger, mockProtocol);
+                try {
+                    let startCompleted = false;
+                    const startPromise = hubConnection.start().then(() => startCompleted = true);
+                    expect(startCompleted).toBe(false);
```

Comment:
Shouldn't this assert after the await?

Reply:
No, it's asserting that `startCompleted` is **false**. It will be `true` after the `await`. This is to ensure that the promise doesn't complete until the handshake is received.